lib/pages/child/rewards_page.dart

```diff
@@ -25,77 +29,81 @@ class _ChildRewardsPageState extends State<ChildRewardsPage> {
-										graphicHeight: 44.0,
-										progressPercentage: 0.1,
-										activeProgressBarColor: AppColors.childActionColor,
-										chips: [
-											AttributeChip.withCurrency(
-												currencyType: CurrencyType.diamond,
-												content: '200',
-											)
-										],
-										actionButton: ItemCardActionButton(
-											disabled: true,
-											color: AppColors.childActionColor,
-											icon: Icons.add_box,
-											onTapped: () => showRewardDialog(context, UIReward(name: 'Zakupomania', icon: 13, cost: UIPoints(quantity: 900, type: CurrencyType.diamond)))
+									if((state as ChildRewardsLoadSuccess).claimedRewards.isNotEmpty)
+										Segment(
+											title: '$_pageKey.claimedRewardsTitle',
+											elements: _buildRewardHistory(state)
 										)
-									)
 								]
 							),
-							// TODO Show only if there are any claimed rewards
-							Segment(
-								title: '$_pageKey.claimedRewardsTitle',
-								elements: <Widget>[
-									ItemCard(
-										title: '1 godzina gry na konsoli',
-										subtitle: 'Odebrano 09.08.2020',
-										graphic: 9,
-										graphicType: AssetType.rewards,
-										graphicHeight: 44.0,
-										isActive: false
-									)
-								]
-							)
-						]
-					)
+						wrapWithExpanded: true,
+		      )
 				]
 			),
 			bottomNavigationBar: AppNavigationBar.childPage(currentIndex: 1)
 		);
 	}
+
+	void _claimReward(UIReward reward) {
+		context.bloc<ChildRewardsCubit>().claimReward(reward);
+		Navigator.of(context).pop(); // closing confirm dialog before pushing snackbar
+		showSuccessSnackbar(context, '$_pageKey.rewardClaimedText');
+	}
+
+	List<Widget> _buildRewardShop(ChildRewardsLoadSuccess state, BuildContext context) {
+		return state.rewards.map((reward) {
+			double percentage = (state.points[reward.cost.type] ?? 0) / reward.cost.quantity; 
+			return ItemCard(
+				title: reward.name,
+				graphic: reward.icon,
+				graphicType: AssetType.rewards,
+				graphicHeight: 44.0,
+				progressPercentage: percentage >= 1.0 ? 1.0 : percentage,
+				activeProgressBarColor: AppColors.currencyColor[reward.cost.type],
+				chips: [
+					AttributeChip.withCurrency(
+						currencyType: reward.cost.type,
+						content: reward.cost.quantity.toString(),
+						tooltip: '$_pageKey.claimCostLabel'
+					)
+				],
+				actionButton: ItemCardActionButton(
+					color: AppColors.currencyColor[reward.cost.type],
+					icon: Icons.add_shopping_cart,
+					disabled: percentage < 1.0,
+					onTapped: () => showRewardDialog(context, reward, claimFeedback: () => _claimReward(reward))
+				)
+			);
+		}).toList();
+	}
+
+	List<Widget> _buildRewardHistory(ChildRewardsLoadSuccess state) {
+		return state.claimedRewards.map((reward) {
+			return ItemCard(
+				title: reward.name,
+				subtitle: AppLocales.of(context).translate('$_pageKey.claimDateLabel') + ' ' +
+					DateFormat.yMd(Localizations.localeOf(context).toString()).format(reward.date).toString(),
```

Comment:
This works now but will become invalid with my manual locale changing merge - instead use `AppLocales.instance.locale`